### PR TITLE
Deprecations and spec updates post-0.12

### PR DESCRIPTION
**Pull request by rustyrussell**

### CODE REVIEW

**Comment by rustyrussell** [comment]
Rebased, fixed one place in tests where I missed "WIRE_FUNDING_LOCKED"->"WIRE_CHANNEL_READY".

**Review by vincenzopalazzo — APPROVED** [review]
ACK [380ee93](https://github.com/ElementsProject/lightning/pull/5490/commits/380ee93833110cb72fd0c8b9599935a816d97063)

File: lightningd/jsonrpc.c [hunk 1]
```diff
@@ -709,14 +709,10 @@ static void replace_command(struct rpc_command_hook_payload *p,
...
-	// from this more restrictive change.
-	if (!deprecated_apis) {
-		const jsmntok_t *jsonrpc = json_get_member(buffer, replacetok, "jsonrpc");
-		if (!jsonrpc || jsonrpc->type != JSMN_STRING || !json_tok_streq(buffer, jsonrpc, "2.0")) {
-			bad = "jsonrpc: \"2.0\" must be specified in the request";
-			goto fail;
-		}
+	jsonrpc = json_get_member(buffer, replacetok, "jsonrpc");
+	if (!jsonrpc || jsonrpc->type != JSMN_STRING || !json_tok_streq(buffer, jsonrpc, "2.0")) {
+		bad = "jsonrpc: \"2.0\" must be specified in the request";
```
Comment (Sjors) [inline lightningd/jsonrpc.c:714]: I think this broke the BTCPay server integration… cc @NicolasDorier

Comment (vincenzopalazzo) [reply lightningd/jsonrpc.c:714]: The broken part is the RPC library used inside it

**Pull request merged**